### PR TITLE
Remove super errors

### LINKED ISSUE #24: Remove super-errors
rationale:

- hard to maintain (depends on copied code from upstream OCaml)
- slow to build 
- depends on patches upstream too, which we can remove
- errors in 4.12 are just much better than before.

We should probably be careful to keep the option alive for compatibility with folks that have set `-bs-super-errors` in their `bsc-flags`

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes #24

### CODE REVIEW

File: esy.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "@opam/merlin": "*"
   },
   "resolutions": {
-    "ocaml": "anmonteiro/ocaml#72babec0e1796ce322464afc8ef2a0d7125d60cb"
+    "ocaml": "anmonteiro/ocaml#75f22c872451c66a4c4aadc43abee55697268a57"
```
Comment (anmonteiro) [inline esy.json:16]: CALLOUT: updated the OCaml compiler hash, where I reverted the super error commits.

File: jscomp/main/js_main.ml [hunk 2]
```diff
@@ -289,7 +287,8 @@ let buckle_script_flags : (string * Bsc_args.spec * string) array =
     (******************************************************************************)
 
 
-    "-bs-super-errors", unit_lazy Super_main.setup,
+    (* XXX(anmonteiro): flag kept for compatibility. *)
```
Comment (Mathspy) [inline jscomp/main/js_main.ml:290]: Never seen this comment syntax before, what does `XXX` mean?

Comment (anmonteiro) [reply jscomp/main/js_main.ml:290]: Leaving this SO answer here for posterity: https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments

I use it in the sense of "heads up, this needs attention".

Comment (Mathspy) [reply jscomp/main/js_main.ml:290]: Ahaa, neat! Do you sign your name between the parentheses as the person who added it or is it “look here, future @anmonteiro)?

Comment (anmonteiro) [reply jscomp/main/js_main.ml:290]: it's the person who added, so I can know who to bother in the future. In this case, my past self :)

Comment (Mathspy) [reply jscomp/main/js_main.ml:290]: Gotcha
Thumbs Up but weeb
Feel free to mark as resolved or keep it open to be easier to notice by others ❤️

**Review by EduardoRFS — APPROVED** [review]
anime girl gif having fun

**Pull request merged**